app/Http/Controllers/API/DeleteAccountController.php

```diff
@@ -0,0 +1,323 @@
+                ->get();
+
+            if (sizeof($states) > 0)
+            {
+                foreach ($states as $state)
+                {
+                    $state->created_by = null;
+                    $state->user_id_last_uploaded = null;
+                    $state->save();
+                }
+            }
+
+            $cities = City::select('id', 'created_by', 'user_id_last_uploaded')
+                ->where('created_by', $userId)
+                ->orWhere('user_id_last_uploaded', $userId)
+                ->get();
+
+            if (sizeof($cities) > 0)
+            {
+                foreach ($cities as $city)
+                {
+                    $city->created_by = null;
+                    $city->user_id_last_uploaded = null;
+                    $city->save();
+                }
+            }
+
+            $littercoin = Littercoin::where('user_id', $userId)->get();
+
+            if (sizeof($littercoin) > 0)
+            {
+                foreach ($littercoin as $ltc)
+                {
+                    $ltc->delete();
+                }
+            }
+
+            $modelHasRoles = DB::table('model_has_roles')
+                ->where('model_type', 'App\Models\User\User')
+                ->where('model_id', $userId)
+                ->get();
+
+            if (sizeof($modelHasRoles) > 0)
+            {
+                foreach ($modelHasRoles as $modelHasRole)
+                {
+                    $modelHasRole->delete();
+                }
+            }
+
+            $oauthTokens = DB::table('oauth_access_tokens')
+                ->where('user_id', $userId)
+                ->get();
+
+            if (sizeof($oauthTokens) > 0)
+            {
+                foreach ($oauthTokens as $oauthToken)
+                {
+                    $oauthToken->delete();
+                }
+            }
+
+            // payments
+            $payments = Payment::where('user_id', $userId)->get();
+
+            if (sizeof($payments) > 0)
+            {
+                foreach ($payments as $payment)
+                {
+                    // should we do something on stripe with stripe_id?
+                    $payment->user_id = 1;
+                    $payment->save();
+                }
+            }
+
+            // photos
+            // subscriptions
+            $subscriptions = DB::table('subscriptions')
+                ->where('user_id', $userId)
+                ->get();
+
+            if (sizeof($subscriptions) > 0 )
+            {
+                foreach ($subscriptions as $subscription)
+                {
+                    $subscription->delete();
+                }
+            }
+
+            // team_user
+            $teamUsers = DB::table('team_user')
+                ->where('user_id', $userId)
+                ->get();
+
+            foreach ($teamUsers as $teamUser)
+            {
+                $teamUser->delete();
+            }
+
+            $teams = Team::where('leader', $userId)
```

Comment:
If the team has no other member we delete the team. If the team has other members, we assign the first one as the leader.
Also, I think we need to split where and orWhere for leader and created_by and do different actions for them.